### ISSUE TITLE
Factorise component

### ISSUE THREAD

**Issue by Blightwidow** [AlbertWhite/apollo-fullstack-todolist#5]
Instead of having boolean flags or a switch/case in the component, you could just have a `onAdd` and `children` props that would allow you to create as many `AddButton` as you need by allowing the parent to handle it. 

If you don't want the parent to become too big because of this you could create two layers: A generic `AddButton` that take children and `onAdd` function then two components `AddUserButton` and `AddTodoButton` that feed the lower component with the correct handler and message. This allow to have a single source of truth on html/css of the button while not putting every behavior in the parent

https://github.com/AlbertWhite/apollo-fullstack-todolist/blob/0a35b32b4a05ee20551a8fea853a3a375562bb5d/client/src/components/Add.tsx#L13

**Comment by AlbertWhite** [comment]
Thanks a lot for your suggestion !

**Comment by AlbertWhite** [comment]
I agree with your proposition. The old way of my code is adding complexités to the shared component, in this way we must modify this shared component everytime when we want to add new case.
The way you proposed is better: we make the shared component unchangeable, and we pass text by children props and pass callback function in props.

**Comment by AlbertWhite** [comment]
Hi Théo, I made a commit for refactoring the Add Component.
The only part that I am not sure is with 'useCallback' 

In file AddTodo.tsx
```js
  const onAdd = useCallback(
    (input: React.MutableRefObject<HTMLInputElement>): void => {
      addTodo({
        variables: {
          content: input.current!.value || 'default name',
          userId
        }
      })
      input.current!.value = 'default name'
    },
    []
  )

  return <Add onAdd={onAdd}>Add Todo</Add>
```

I am not sure if I should use useCallback here.
